File: 20171215/p06_random_test.c
//-------------------------------------------------------------------------
//-- PHYS404/604 (2013)
//- Program: test of random number 
//--------------------------------------------------------------------------

#include <stdio.h>
#include <math.h>

double random();

int main(void)
{
  int nc = 10;
  double r,count[nc];
  int i,irand,nrand,seed=1;
  FILE *fp1,*fp2;

  // number of random number
  nrand = 10000;

  // clear count array
  for(i=0;i<nc;i++) count[i]=0.0;

  fp1=fopen("rand.dat","w");
  fp2=fopen("count.dat","w");


  for(i=0;i<=nrand;i++){
    // uniform random number generator [0.0,1.0)
    r = random(&seed);
    fprintf(fp1,"%d  %16.8e\n",i,r);

    // counter (spectrum)
    irand = (int)(r*nc);

    count[irand] = count[irand] + 1;
  }

  for(i=0;i<nc;i++){
    fprintf(fp2,"%16.8e  %16.8e\n",(double)i/nc,count[i]);
  }

  fclose(fp1);
  fclose(fp2);
}

File: 20171215/p06_distrib/p06_random_test.f90
!--------------------------------------------------------------------------
!-- Program: test of random number 
!--------------------------------------------------------------------------
  implicit none
  real(kind=8)	:: r,random,count(0:9)
  integer i,j,seed,irand

! initial random seed (non zero)
  seed =1

! clear count array
count=0.0

! open an output file with IO=10, name: err.dat
  open(10,file='rand.dat')
  open(11,file='count.dat')

  do i=0,10000
     r = random(seed)
     write(10,'(i6,e16.8)') i,r
     irand = r*10
     count(irand) = count(irand) + 1
  enddo

  do j=0,9
     write(11,'(2e16.8)') j*0.1,count(j)
  enddo

! close the output file
  close(10)
  close(11)

  stop
end program
